integration-tests/bats/filter-branch.bats

```diff
@@ -400,5 +400,30 @@ SQL
     [[ "$output" =~ "does not exist" ]] || false
 }
 
+@test "filter-branch: fails with working and staged changes" {
```

Comment:
I think we should test the all branch head case as well.

Reply:
https://docs.dolthub.com/cli-reference/cli#dolt-filter-branch

See `--all` and `--branches`.

Reply:
Also, maybe make it respect `--continue` and just blow away the working and staged like today?

Reply:
Actually, that last one is a bad idea. Ignore that :-)